src/sinks/loki/service.rs

```diff
@@ -94,6 +95,9 @@ impl Service<LokiRequest> for LokiService {
         if let Some(tenant_id) = request.tenant_id {
             req = req.header("X-Scope-OrgID", tenant_id);
         }
+        if request.is_compressed {
+            req = req.header("Content-Encoding", "gzip");
+        }
```

Comment:
Rather than hardcoding this to `gzip` this should call the `content_encoding` function on `Compression`:

https://github.com/vectordotdev/vector/blob/708f597db18a81a1f908b09e5c7654a9a1ff9352/src/sinks/util/buffer/compression.rs#L39-L44